File: zentrapay_application/lib/core/models/card.dart
class AppCard {
  final String cardId;
  final String brand;
  final String cardType;
  final String last4;
  final int expiryMonth;
  final int expiryYear;
  final bool nfcEnabled;
  final bool qrEnabled;
  final String status;
  final String balance;
  final String currencyCode;
  final String createdAt;

  AppCard({
    required this.cardId,
    required this.brand,
    required this.cardType,
    required this.last4,
    required this.expiryMonth,
    required this.expiryYear,
    required this.nfcEnabled,
    required this.qrEnabled,
    required this.status,
    required this.balance,
    required this.currencyCode,
    required this.createdAt,
  });

  factory AppCard.fromJson(Map<String, dynamic> json) => AppCard(
    cardId: json['cardId'] ?? '',
    brand: json['brand'] ?? '',
    cardType: json['cardType'] ?? 'VIRTUAL',
    last4: json['last4'] ?? '0000',
    expiryMonth: json['expiryMonth'] ?? 1,
    expiryYear: json['expiryYear'] ?? 0,
    nfcEnabled: json['nfcEnabled'] ?? false,
    qrEnabled: json['qrEnabled'] ?? false,
    status: json['status'] ?? 'active',
    balance: json['balance'] ?? '0.0000',
    currencyCode: json['currencyCode'] ?? "unknown",
    createdAt: json['createdAt'],
  );

  AppCard copyWith({bool? nfcEnabled, bool? qrEnabled}) => AppCard(
    cardId: cardId,
    brand: brand,
    cardType: cardType,
    last4: last4,
    expiryMonth: expiryMonth,
    expiryYear: expiryYear,
    nfcEnabled: nfcEnabled ?? this.nfcEnabled,
    qrEnabled: qrEnabled ?? this.qrEnabled,
    status: status,
    balance: balance,
    currencyCode: currencyCode,
    createdAt: createdAt,
  );
}

File: zentrapay_spring_boot_layer/src/main/java/com/zentrapay_application/zentrapay_spring_boot_layer/domain/model/Card.java
package com.zentrapay_application.zentrapay_spring_boot_layer.domain.model;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

/**
 * Canonical card — consolidates the old cards.CardsModel and
 * zpay.CardModel (both mapped "cards" with different, incompatible columns).
 */
@Entity
@Data
@Table(name = "cards")
public class Card {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "card_id", nullable = false)
    private UUID cardId;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(name = "wallet_id")
    private UUID walletId;

    @Column(nullable = false, length = 30)
    private String brand;

    @Column(name = "card_type", nullable = false, length = 20)
    private String cardType = "VIRTUAL"; // VIRTUAL, PHYSICAL

    @Column(nullable = false, length = 4)
    private String last4;

    @Column(name = "expiry_month", nullable = false)
    private short expiryMonth;

    @Column(name = "expiry_year", nullable = false)
    private short expiryYear;

    @Column(name = "nfc_enabled", nullable = false)
    private boolean nfcEnabled = true;

    @Column(name = "qr_enabled", nullable = false)
    private boolean qrEnabled = true;

    @Column(nullable = false, length = 20)
    private String status = "ACTIVE";

    @CreationTimestamp
    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}
